zerver/views/home.py

```diff
@@ -205,6 +209,8 @@ def home_real(request: HttpRequest) -> HttpResponse:
         furthest_read_time    = sent_time_in_epoch_seconds(latest_read),
         has_mobile_devices    = num_push_devices_for_user(user_profile) > 0,
         bot_types             = get_bot_types(user_profile),
+        two_fa_enabled        = settings.TWO_FACTOR_AUTHENTICATION_ENABLED,
+        two_fa_enabled_user   = bool(default_device(user_profile)),
```

Comment:
Is this the line that adds 3 database to loading the home view?  I'm wondering if there's a better way to do this.

Reply:
I have added a little optimization. It's not perfect but will save us the 3 queries when 2FA is not enabled.